### PR TITLE
Add unix-type-representations 0.1.2

### PR DESCRIPTION
**Pull request by yallop**

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
For some reason the tests fail on Alpine:
```
#=== ERROR while compiling unix-type-representations.0.1.2 ====================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.14.0 | pinned(https://github.com/yallop/ocaml-unix-type-representations/archive/0.1.2.tar.gz)
# path                 ~/.opam/4.14/.opam-switch/build/unix-type-representations.0.1.2
# command              ~/.opam/opam-init/hooks/sandbox.sh build ocaml pkg/pkg.ml test
# exit-code            1
# env-file             ~/.opam/log/unix-type-representations-3162-27330d.env
# output-file          ~/.opam/log/unix-type-representations-3162-27330d.out
### output ###
# .....E
# ==============================================================================
# Error: Unix type representation tests:5:use dir_handle with ctypes.
# 
# File "/home/opam/.opam/4.14/.opam-switch/build/unix-type-representations.0.1.2/_build/oUnit-Unix type representation tests-builder#00.log", line 86, characters 1-1:
# Error: Unix type representation tests:5:use dir_handle with ctypes (in the log).
# 
# Worker stops running: Killed by signal -10
# ------------------------------------------------------------------------------
# Ran: 6 tests in: 1.25 seconds.
# FAILED: Cases: 6 Tried: 6 Errors: 1 Failures: 0 Skip:  0 Todo: 0 Timeouts: 0.
# pkg.ml: [ERROR] Some tests failed.
```

File: packages/unix-type-representations/unix-type-representations.0.1.2/opam [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+maintainer: "Jeremy Yallop <yallop@gmail.com>"
+authors: "Jeremy Yallop <yallop@gmail.com>"
+license: "MIT"
+homepage: "https://github.com/yallop/ocaml-unix-type-representations"
+bug-reports: "https://github.com/yallop/ocaml-unix-type-representations/issues"
+dev-repo: "git+https://github.com/yallop/ocaml-unix-type-representations.git"
+build: [
+  ["ocaml" "pkg/pkg.ml" "build" "--dev-pkg" "%{pinned}%"]
+  ["ocaml" "pkg/pkg.ml" "build" "--dev-pkg" "%{pinned}%" "--tests" "true"]
+    {with-test}
```
Comment (kit-ty-kate) [inline packages/unix-type-representations/unix-type-representations.0.1.2/opam:11]:
```suggestion
  ["ocaml" "pkg/pkg.ml" "build" "--dev-pkg" "%{dev}%"]
  ["ocaml" "pkg/pkg.ml" "build" "--dev-pkg" "%{dev}%" "--tests" "true"]
    {with-test}
```

File: packages/unix-type-representations/unix-type-representations.0.1.2/opam [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  "ocaml"
+  "topkg" {build & >= "0.9.0"}
+  "ocamlfind" {build}
+  "ocamlbuild" {build}
+  "base-unix"
+  "ctypes" {with-test}
+  "ounit" {with-test}
+]
+synopsis:
+  "Functions that expose the underlying types of some abstract types in the Unix module."
```
Comment (kit-ty-kate) [inline packages/unix-type-representations/unix-type-representations.0.1.2/opam:24]:
```suggestion
  "Functions that expose the underlying types of some abstract types in the Unix module"
```

**Comment by yallop** [comment]
> For some reason the tests fail on Alpine

I can't easily investigate at the moment, so I've added an `available` constraint (936398a) and opened an issue (https://github.com/yallop/ocaml-unix-type-representations/issues/7).

File: packages/unix-type-representations/unix-type-representations.0.1.2/opam [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  ["ocaml" "pkg/pkg.ml" "build" "--dev-pkg" "%{dev}%"]
+  ["ocaml" "pkg/pkg.ml" "build" "--dev-pkg" "%{dev}%" "--tests" "true"]
+    {with-test}
+  ["ocaml" "pkg/pkg.ml" "test"] {with-test}
+]
+depends: [
+  "ocaml"
+  "topkg" {build & >= "0.9.0"}
+  "ocamlfind" {build}
+  "ocamlbuild" {build}
```
Comment (kit-ty-kate) [inline packages/unix-type-representations/unix-type-representations.0.1.2/opam:18]:
```suggestion
  "ocamlbuild" {build & != "0.9.0"}
```
```
#=== ERROR while compiling unix-type-representations.0.1.2 ====================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.03.0 | pinned(https://github.com/yallop/ocaml-unix-type-representations/archive/0.1.2.tar.gz)
# path                 ~/.opam/4.03/.opam-switch/build/unix-type-representations.0.1.2
# command              ~/.opam/opam-init/hooks/sandbox.sh build ocaml pkg/pkg.ml build --dev-pkg false
# exit-code            1
# env-file             ~/.opam/log/unix-type-representations-1324-a9cef3.env
# output-file          ~/.opam/log/unix-type-representations-1324-a9cef3.out
### output ###
# ocamlfind ocamlopt unix.cmxa -I /home/opam/.opam/4.03/lib/ocamlbuild /home/opam/.opam/4.03/lib/ocamlbuild/ocamlbuildlib.cmxa -linkpkg myocamlbuild.ml /home/opam/.opam/4.03/lib/ocamlbuild/ocamlbuild.cmx -o myocamlbuild
# ocamlfind: Package `ctypes' not found
# ocamlfind ocamlc -g -I '' -c -o lib/unix_type_representation_stubs.o lib/unix_type_representation_stubs.c
# ocamlfind ocamlmklib -o lib/unix_type_representations_stubs lib/unix_type_representation_stubs.o
# + ocamlfind ocamlmklib -o lib/unix_type_representations_stubs lib/unix_type_representation_stubs.o
# gcc: error: lib/unix_type_representation_stubs.o: No such file or directory
# gcc: fatal error: no input files
# compilation terminated.
# Command exited with code 2.
# pkg.ml: [ERROR] cmd ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-tag' 'debug'
#      '-build-dir' '_build' 'opam' 'pkg/META' 'CHANGES.md' 'LICENSE'
#      'README.md' 'lib/unix_type_representations.a'
#      'lib/unix_type_representations.cmxs'
#      'lib/unix_type_representations.cmxa' 'lib/unix_type_representations.cma'
#      'lib/unix_representations.cmx' 'lib/unix_representations.cmi'
#      'lib/unix_representations.mli' 'pkg/META' 'README.md'
#      'lib/dllunix_type_representations_stubs.so'
#      'lib/libunix_type_representations_stubs.a']: exited with 10
```

**Comment by kit-ty-kate** [comment]
Thanks. Could you return the fixes in the opam file upstream?

**Comment by yallop** [comment]
Done (https://github.com/yallop/ocaml-unix-type-representations/commit/51c871ebaa84feb0ea6dc9fd12b708bd00fc429d). Thanks!

**Pull request merged**